easyDiffractionApp/Logic/PyQmlProxy.py

```diff
@@ -1397,3 +1514,74 @@ def _onStatusInfoChanged(self):
     @Property('QVariant', notify=dummySignal)
     def screenRecorder(self):
         return self._screen_recorder
+
+    ####################################################################################################################
+    ####################################################################################################################
+    # Undo/Redo stack operations
+    ####################################################################################################################
+    ####################################################################################################################
+
+    @Property(bool, notify=undoRedoChanged)
+    def canUndo(self) -> bool:
+        return borg.stack.canUndo()
+
+    @Property(bool, notify=undoRedoChanged)
+    def canRedo(self) -> bool:
+        return borg.stack.canRedo()
+
+    @Slot()
+    def undo(self):
+        if self.canUndo:
+            callback = [self.parametersChanged]
```

Comment:
just a stylistic comment - early exits look nicer in cases where the whole body of a method depends on a conditional.
i.e.
```
if not self.canUndo:
    return

( body of the method without additional indentation)
```